src/main/java/org/jenkinsci/test/acceptance/Matchers.java

```diff
@@ -77,6 +78,20 @@ public void describeMismatchSafely(WebDriver item, Description d) {
         };
     }
 
+    public static Matcher<WebDriver> hasURL(final URL url) {
```

Comment:
@olivergondza I used URL because the TopLevelItems url() method returns a URL, so it's quite straightforward to do something like waitFor(driver, hasURL(jenkins.url), 2).